CentralComputing/UDPManager.cpp

```diff
@@ -77,38 +77,37 @@ bool UDPManager::start_udp(const char * hostname, const char * send_port, const
   return false;
 }
 
-void UDPManager::udp_recv(){
-  uint8_t recv_buf[16];
-  socklen_t fromlen; // needed for recvfrom
-  struct sockaddr_storage fromaddr;// needed for recvfrom
-  fromlen = sizeof fromaddr; // needed for recvfrom
+int UDPManager::udp_recv(uint8_t* recv_buf, uint8_t len){
+  
+  socklen_t fromlen;                // needed for recvfrom
+  struct sockaddr_storage fromaddr; // needed for recvfrom
+  fromlen = sizeof fromaddr;        // needed for recvfrom
 
-  int byte_count = recvfrom(socketfd, recv_buf, sizeof recv_buf, 0, 
+  int byte_count = recvfrom(socketfd, recv_buf, len, 0, 
       (struct sockaddr *)&fromaddr, &fromlen);
-  //int byte_count = recv(socketfd, recv_buf, sizeof recv_buf,0);
 
   recv_buf[byte_count] = '\0';
   print(LogLevel::LOG_DEBUG, "recv %d bytes, they are: %s \n", byte_count, recv_buf);
+  
+  udp_parse(recv_buf, len);
+
+  return byte_count;
+}
 
-  //TODO: Parse the recv_buf. 
-  //TODO: use byte_count to know how big the read was
-  //
-  //TODO: return _something_. Change the signature if necessary
+void UDPManager::udp_parse(uint8_t* buf, uint8_t len){
```

Comment:
An idea for what this function should do:

We call this function, and 99% of the time we expect it to parse a ping. If it was a ping, we want to let our timer/logic/ whatever know that the Python server is still alive.

On the odd case that we get sent something else, then this function can parse it, and add the command to the unified command queue (once it gets implemented).

Here is an idea: have it return `true` if it was a ping, false otherwise. let this function handle any other odd command.

Reply:
So was thinking this function should basically check to see if its a ping (return true), else parse it and put it on the unified command queue (return false). 

From there, we can use the return value to confirm it received our command. (instead of it sending "ack" it sends back "kca")? (Not sure if we'd need to do this, it seems like something maybe the TCP server would do).